exif, tiff: Simplify and add comments.

mewmew opened the pull request:
- Add comment about marker value, EXIF (0xE1 = APP1).
- Simplify newAppSec.
- Use io.ReadFull where appropriate.
- Use *bytes.Buffer for *Tiff String() method.

rwcarlsen commented on exif/exif.go:
```diff
@@ -184,25 +185,25 @@ func newAppSec(marker byte, r io.Reader) (*appSec, error) {
 			continue
 		}
 
-		dataLenBytes, err := br.Peek(2)
```
You created a bug here because dataLen _includes_ the two bytes that make the dataLen (whence the peek).  Your changes cause the loop that reads the section data to read 2 bytes past the end of the section.  Most of the changes in this pull request look alright - except these ones in the newAppSec function don't particularly seem to add much other than just doing things a slightly different way.

rwcarlsen replied:
This pull request looks generally good - if you revert newAppSec to its prior state, it's worth a merge.

Commit:
exif, tiff: Simplify and add comments.

- Add comment about marker value, EXIF (0xE1 = APP1).
- Simplify newAppSec.
- Use io.ReadFull where appropriate.
- Use *bytes.Buffer for *Tiff String() method.

rwcarlsen commented:
Preferrably in a single commit (via reset + "add -i" or rebase) - so newAppSec doesn't wobble back and forth between two separate commits.  Or I can do it if you don't mind "losing credit" for the contribution in the repo history (I could throw your username in the commit message).

mewmew commented:
Thank you for spotting the bug I introduced. I've closed this pull request in favor of pull request 10.
